src/main/java/com/greghaskins/spectrum/Spectrum.java

```diff
@@ -314,11 +276,46 @@ public static void aroundEach(ThrowingConsumer<com.greghaskins.spectrum.Block> c
    * @param consumer to run each spec block
    */
   public static void aroundAll(ThrowingConsumer<com.greghaskins.spectrum.Block> consumer) {
-    getCurrentSuiteBeingDeclared().aroundAll(consumer);
+    addHook(new HookContext(consumer::acceptOrThrow, getDepth(),
+        HookContext.AppliesTo.ONCE, HookContext.Precedence.OUTER));
   }
 
+  /**
+   * A value that will be fresh within each spec and cannot bleed across specs.
+   *
+   * <p>
+   * Note that {@code let} is lazy-evaluated: the {@code supplier} is not called until the first
+   * time it is used.
+   * </p>
+   *
+   * @param <T> The type of value
+   *
+   * @param supplier {@link com.greghaskins.spectrum.ThrowingSupplier} function that either
+   *        generates the value, or throws a `Throwable`
+   * @return supplier which is refreshed for each spec's context
+   */
+  public static <T> Supplier<T> let(final com.greghaskins.spectrum.ThrowingSupplier<T> supplier) {
+    LetHook<T> letHook = new LetHook<>(supplier);
+    HookContext hookContext = new HookContext(letHook, getDepth(),
+        HookContext.AppliesTo.EACH_CHILD, HookContext.Precedence.LOCAL);
+    addHook(hookContext);
+
+    return letHook;
+  }
+
+  /**
+   * Will throw an exception if this method happens to be called while Spectrum is still defining
+   * tests, rather than executing them. Useful to see if a hook is being accidentally used
+   * during definition.
+   */
+  public static void assertSpectrumInTestMode() {
+    if (getCurrentSuiteBeingDeclared() != null) {
+      throw new IllegalStateException("Cannot use this statement in a suite declaration. "
+          + "It may only be used in the context of a running spec.");
+    }
+  }
 
-  private static final Deque<Suite> suiteStack = new ArrayDeque<>();
+  private static final Variable<Deque<Suite>> suiteStack = new Variable<>(ArrayDeque::new);
```

Comment:
It's a threadsafe `Deque` - I guess this is itself just a wrapper for `ThreadLocal` but it feels nice using a Spectrum construct.